Changes to the title column of the item table.

bendk opened the pull request:
Before title tracked the title that we got from metadata (file tags, echonest,
etc).  Now that's stored in metadata_title, and title is the computed value of
title (what get_title() used to return).

This is needed if we are going to implement frontend item list using SQL
queries, since if the user sorts by title, we need this for the ORDER BY
clause.

I wrote this while working on a private branch to implement the plan in FrontendDataOverhaul.  I was thinking that maybe it would be a good idea to commit changes like this to master that are self-contained DB schema changes.  I think it might make merging less painful when I'm done with everything.

I could see an argument to not checking this in until I was done with all my work for FrontendDataOverhaul.  If you want to make an argument for that, then I could wait on merging.

Commit:
Changes to the title column of the item table.

Before title tracked the title that we got from metadat (file tags, echonest,
etc).  Now that's stored in metadata_title, and title is the computed value of
title (what get_title() used to return).

This is needed if we are going to implement frontend item list using SQL
queries, since if the user sorts by title, we need this for the ORDER BY
clause.

paulswartz commented on tv/lib/item.py:
```diff
@@ -995,7 +997,11 @@ def file_moved(self, new_filename):
         """Update our attributes from a metadata dictionary."""
+        # change the name of title to be "metadata_title"
+        if 'title' in metadata_dict:
+            metadata_dict['metadata_title'] = metadata_dict['title']
```
Should this be `.pop('title')`?

bendk replied:
Good point.  Pushed a new commit for this.

Commit:
Use dict.pop() to prevent duplicate entries.

Thanks z3p for pointing this one out.

Commit:
Work with a copy instead of modifying the metadata dict

I think either would have worked with the current system, but it seems cleaner
not to modify the dict we're passed in.  This also fixes the unittests.
